src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -20,15 +20,25 @@ client_pool::client_pool(
   , _policy(policy) {}
 
 ss::future<> client_pool::stop() {
+    vlog(pool_log.info, "Stopping client pool: {}", _pool.size());
+
     if (!_as.abort_requested()) {
         _as.request_abort();
     }
     _cvar.broken();
     // Wait until all leased objects are returned
     co_await _gate.close();
+
+    for (auto& it : _pool) {
+        co_await it->stop();
+    }
+
+    vlog(pool_log.info, "Stopped client pool");
 }
 
 void client_pool::shutdown_connections() {
+    vlog(pool_log.info, "Shutting down client pool: {}", _pool.size());
```

Comment:
is `info` level excessive? is this a frequent thing? (same for start/stop)

Reply:
They only happen on application shut-down. This change will make shutdown wait until Azure terminates the connection, so I went with info level to be able to debug potential hangs. I can change it if you think it's excesive.

Reply:
thanks